[TSN]: update linuxptp and other ethernet tools for Time Sensitive Networking

aaroncyew opened the pull request:
#### Merge Checklist  <!-- REQUIRED -->
**All** boxes should be checked before merging the PR
- [x] The changes in the PR have been built and tested
- [x] cgmanifest file has been updated if required
- [] Ready to merge

#### Description <!-- REQUIRED -->
<!-- Please include a summary of the changes and the related issue. List any dependencies that are required for this change. -->
Updated (5) specs - linuxptp, ethtool, xdp-tools, libbpf, iproute
Added TSN patches for the (5) specs 
Updated signatures for linuxptp v3.1.1 to v4.3
Updated cgmanifest for linuxptp source changes
<!-- Fixes # (issue) -->

#### Any Newly Introduced Dependencies
<!-- Please describe any newly introduced 3rd party dependencies in this change. List their name, license information and how they are used in the project. -->
No new dependencies being introduce, reuse existing versions of specs for xdp-tools, iproute, ethtool, libbpf

#### How Has This Been Tested? <!-- REQUIRED -->
<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

aaroncyew committed:
Update linuxptp v4.3

- add Time Sensitive Networking(TSN) patches
- remove linuxptp-zerolength.patch, patch is merged in linuxptp v4.3
- remove enable-ha.patch, patch not needed for TSN

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

aaroncyew committed:
Update xdp-tools rev3 to support TSN

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

aaroncyew committed:
Update ethtool to rev3 to support TSN

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

aaroncyew committed:
Update iproute to rev3 to support TSN

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

aaroncyew committed:
update libbpf to rev2 to support TSN

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

aaroncyew committed:
Update linuxptp sources on cgmanifest

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

sgolebiewski-intel committed:
Updating Readme file - porting #177 (#183)

magerstam committed:
added proxy instruction for dnf and tdnf (#184)

* added proxy instruction for dnf and tdnf

* Update docs/developer-guide/emt-troubleshooting.md

Co-authored-by: Sebastian Golebiewski <sebastianx.golebiewski@intel.com>

* Update docs/developer-guide/emt-troubleshooting.md

Co-authored-by: Sebastian Golebiewski <sebastianx.golebiewski@intel.com>

---------

Co-authored-by: Sebastian Golebiewski <sebastianx.golebiewski@intel.com>

rranjan3 committed:
Update CA/HDA/PTA/PUA versions for grpc retry (#182)

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

sfonn committed:
Add openbox windows manager for EMF-X11 (#142)

* Add openbox windows manager

Initial import of openbox windows manager
Add openbox into xorg-x11-packages defn

Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>

* openbox: fix changelog so license gets picked up

---------

Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>
Co-authored-by: Anuj Mittal <anuj.mittal@intel.com>

sfonn committed:
[AMT] Include AMT vPRO in EMT (#170)

* [AMT] Add ACE-TAO rpm

Add ACE-TAO for LMS dependency

Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>

* [AMT] LMS module

Add LMS for AMT support

Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>

* [AMT] Include AMT in EMT image

Add vPRO AMT support into EMT images

Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>

---------

Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>

anujm1 commented on SPECS/libbpf/0001-libbpf-add-txtime-field-in-xdp_desc-struct.patch:
```diff
@@ -0,0 +1,28 @@
+ 	__u64 addr;
+ 	__u32 len;
+ 	__u32 options;
++	__u64 txtime;
```
Could you please make sure we have the corresponding kernel patch?

aaroncyew replied:
this patch is for libbpf, there is no corresponding kernel patch.

anujm1 replied:
This header aligns with kernel and corresponding change would need to be in kernel as well.

HongAun-Intel replied:
This is the corresponding commit in the kernel to match the header.

https://github.com/intel-innersource/os.linux.kernel.mainline-tracking-staging/pull/936/commits/959a5036cbbaacd9d3f36be77bf6e16164ba6883

anujm1 replied:
Thanks, but we don't use this kernel. Including the link here for future: https://github.com/intel/linux-intel-lts/commit/90e81216a3f9a2714b3f004e7da40f42904ad52b

aaroncyew replied:
Hi all,
I've check with our EMT3 latest kernel/kernel-rt sources, its including the headers 
include/uapi/linux/if_xdp.h with --> __64 texttime;, line 155.

both kernel/kernel-rt is using the same source tag lts-v6.12.30-emt-250524T131939Z
: https://github.com/intel/linux-intel-lts/blob/lts-v6.12.30-emt-250524T131939Z/include/uapi/linux/if_xdp.h#L155

anujm1 commented:
Hi @aaroncyew can you please include source of these patches in PR description/changelog?

anujm1 commented on SPECS/linuxptp/linuxptp.spec:
```diff
@@ -24,13 +24,24 @@ Source10:	linuxptp-testsuite-%{testsuite_ver}.tar.gz
 
 # fix handling of zero-length messages
-Patch0:		linuxptp-zerolength.patch
+# Patch0:		linuxptp-zerolength.patch
```
why is this being removed?

aaroncyew replied:
linuxptp-zerolength.patch is part of linuxptp v4.3 - [link](https://github.com/richardcochran/linuxptp/blob/v4.3/sk.c#L491)

anujm1 replied:
github.com/richardcochran/linuxptp/commit/9633ab52460f58c92c6daa35e9d24e4ce9c5ab1c

Lets remove the patch from repository as well and delete this and later line where it's getting applied. Please add this in PR description so we know it was removed because the patch is present upstream now.

aaroncyew replied:
Removed from the linuxptp.spec

anujm1 commented on SPECS/linuxptp/linuxptp.spec:
```diff
@@ -24,13 +24,24 @@ Source10:	linuxptp-testsuite-%{testsuite_ver}.tar.gz
 # The following patch is a combination of multiple patches to enable HA in linuxptp
 # https://review.opendev.org/c/starlingx/integ/+/891638
-Patch2:         enable-ha.patch
+# Patch2:         enable-ha.patch
```
why is this being removed?

aaroncyew replied:
there are 47 patches in this one single patch, and a few patches are already part of linuxptp v4.3 while remaining are enable HA. These HA patches are no found in upstream project linuxptp and its not part of mailing list. Hence this patch is applicable for linuxptp v3.1.1 version of AZL contributed by Red Hat only. TSN enabling doesn't require this HA patch

anujm1 replied:
Please remove it from workspace then

aaroncyew replied:
Removed from linuxptp.spec

anujm1 commented on SPECS/linuxptp/linuxptp.spec:
```diff
@@ -107,10 +130,15 @@ PATH=..:$PATH ./run
 %changelog
+* Wed Jun 04 2025 Aaron Chan <aaron.chun.yew.chan@intel.com> - 4.3-1
+- Add TSN patches/support
+- update to 4.3
```
Please add "Initial Edge Microvisor import ..."

aaroncyew replied:
Ok let me add this in

aaroncyew replied:
Updated changelog to initial import to EMT3

anujm1 commented on SPECS/xdp-tools/0002-remove-prefix-version-9999.patch:
```diff
@@ -0,0 +1,30 @@
+         LIBBPF_CFLAGS=$(${PKG_CONFIG} libbpf --cflags)
+         LIBBPF_LDLIBS=$(${PKG_CONFIG} libbpf --libs)
+-        LIBBPF_VERSION=$(${PKG_CONFIG} libbpf --modversion)
++        LIBBPF_VERSION=$(${PKG_CONFIG} libbpf --modversion | sed 's/^9999.//')
```
what is this for? @cheeyanglee

aaroncyew replied:
I think this is not need, but i'll let @cheeyanglee comment on this

anujm1 replied:
Please remove this

aaroncyew replied:
This patch is removed and libbpf.spec is updated

sgolebiewski-intel committed:
Changing the file structure of documentation - porting # 171 (#185)

cheeyanglee committed:
setuplkgtoolchain.sh: prioritize lkg from development url (#187)

also update urls for manifest

Signed-off-by: Lee Chee Yang <chee.yang.lee@intel.com>

rranjan3 committed:
Revert HDA to 1.7.0 from #182 (#188)

* Revert HDA to 1.7.0 from #182

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

* Review : Fix version in signature

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

---------

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

sgolebiewski-intel committed:
Revert "Changing the file structure of documentation - porting # 171" (#192)

* Revert "Changing the file structure of documentation - porting # 171 (#185)"

This reverts commit fe34d72daa49a32bdd66039db6b59a675a558b0a.

* Update docs/developer-guide/index.rst

aaroncyew commented:
> Hi @aaroncyew can you please include source of these patches in PR description/changelog?

the sources of these patches resides in innersource 
e.g https://github.com/intel-innersource/os.linux.ubuntu.iot.debianpkgs.linuxptp/tree/noble/debian/patches

aaroncyew committed:
Merge branch 'open-edge-platform:3.0-dev' into 3.0-dev

aaroncyew committed:
Remove linuxptp v3.1.1 patches

clean up the following patches not need in linuxptp v4.3
- linuxptp-zerolength
- enable-ha used in starlingX only

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

aaroncyew committed:
Remove redundant patch from xdp-tools

- Remove patch 0002-remove-prefix-version-9999.patch
- Cleanup xdp-tools.spec

Signed-off-by: Aaron Chan <aaron.chun.yew.chan@intel.com>

anujm1 committed:
Fix Changelog and license map

anujm1 committed:
Remove ptp4l from signatures

anujm1 committed:
Merge branch '3.0-dev' into 3.0-dev
